src/client/datascience/interactive-window/interactiveWindow.ts

```diff
@@ -243,7 +251,7 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
                 this._controllerReadyPromise.resolve(e.controller);
 
                 // Recreate the kernel ready promise now that we have a new controller
-                this._kernelReadyPromise = this.createKernelReadyPromise();
+                this.ensureKernelReadyPromise();
             },
```

Comment:
This won't 'recreate' it if it's already set. You probably need to clear it here first.